health-services/plan-service/src/main/java/digit/util/UserUtil.java

```diff
@@ -0,0 +1,78 @@
+package digit.util;
+
+import digit.config.Configuration;
+import lombok.extern.slf4j.Slf4j;
+import org.egov.common.contract.request.RequestInfo;
+import org.egov.common.contract.user.UserDetailResponse;
+import org.egov.common.contract.user.UserSearchRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.Collections;
+
+import static digit.config.ServiceConstants.ERROR_WHILE_FETCHING_FROM_USER_SERVICE;
+
+@Slf4j
+@Component
+public class UserUtil {
+
+    private Configuration configs;
+
+    private RestTemplate restTemplate;
+
+    public UserUtil(RestTemplate restTemplate, Configuration configs) {
+        this.restTemplate = restTemplate;
+        this.configs = configs;
+    }
+
+    /**
+     * This method fetches user details from User Service for the provided employeeID
+     *
+     * @param requestInfo request info from the request
+     * @param employeeId  employee id provided in the request
+     * @param tenantId    tenant id from the request
+     */
+    public UserDetailResponse fetchUserDetail(RequestInfo requestInfo, String employeeId, String tenantId) {
```

Comment:
don't pass employeeId, tenantId according to usecase. Use userSearchCriteria in case later we want to add any more criterias.